src/common/utxobased/db/makeProcessor.ts

```diff
@@ -205,22 +205,24 @@ export async function makeProcessor(
 
     async removeUtxos(utxoIds: string[]): Promise<void> {
       return await baselets.utxo(async tables => {
-        const utxos = await tables.utxoById.query('', utxoIds)
+        const utxos = (await tables.utxoById.query('', utxoIds)).filter(
+          utxo => utxo != null
+        ) as IUTXO[]
         await tables.utxoIdsByScriptPubkey.delete(
           '',
-          utxos.map(utxo => (utxo != null ? utxo.scriptPubkey : undefined))
+          utxos.map(utxo => utxo.scriptPubkey)
         )
         await tables.utxoById.delete('', utxoIds)
       })
     },
 
-    async fetchUtxos(args): Promise<IUTXO[]> {
+    async fetchUtxos(args): Promise<Array<IUTXO | undefined>> {
       const { scriptPubkey, utxoIds = [] } = args
       return await baselets.utxo(async tables => {
         if (scriptPubkey != null) {
-          const utxoIdsByScriptPubkey = (
+          const utxoIdsByScriptPubkey: string[] = (
             await tables.utxoIdsByScriptPubkey.query('', [scriptPubkey])
-          ).filter(utxoId => utxoId != null)
+          ).filter(utxoId => utxoId != null) as string[]
 
           utxoIds.push(...utxoIdsByScriptPubkey)
 
```

Comment:
Looks like we have a bug here. Once the type for `utxoIdsByScriptPubkey` has change to the correct type, we must make sure to change `saveUtxo` to first query from `utxoIdsByScriptPubkey`, modify the resultset, then save it back, then second, need to change the query here in `fetchUtxos` to spread on the first element of the resultset.